api/user/schema.graphql

```diff
@@ -15,9 +15,34 @@ extend type Query {
 extend type Mutation {
   logout: Boolean
   login(email: EmailAddress!, password: String!): User
-  signup(email: EmailAddress!, password: String!): User
+  signup(email: EmailAddress!, password: String!): RegisterUserResponse
   forgotPassword(email: EmailAddress!): Boolean
-  changePassword(token: String!, id: ID!, newPassword: String!): User
+  changePassword(
+    token: String!
+    id: ID!
+    newPassword: String!
+  ): ChangePasswordResponse
+}
+
+union ChangePasswordResponse = UserReturned | ExpiredTokenProblem
```

Comment:
@tobiasdiez have a look at the graphql schema

Reply:
Looks good to me, thanks! 

Not sure if the problem interfaces below need to be "types".